Bucketeer/Sources/Internal/Evaluation/EvaluationInteractor.swift

```diff
@@ -21,85 +21,114 @@ extension EvaluationInteractor {
+        let userAttributesUpdated = evaluationStates.userAttributesUpdated
+        let currentEvaluationsId = evaluationStates.currentEvaluationsId
+        let featureTag = evaluationStates.featureTag
         apiClient.getEvaluations(
             user: user,
             userEvaluationsId: currentEvaluationsId,
-            timeoutMillis: timeoutMillis) { [weak self] result in
-                switch result {
-                case .success(let response):
-                    let newEvaluationsId = response.userEvaluationsId
-                    if currentEvaluationsId == newEvaluationsId {
-                        logger?.debug(message: "Nothing to sync")
-                        completion?(result)
-                        return
-                    }
-                    let newEvaluations = response.evaluations.evaluations
-                    do {
-                        try evaluationDao.deleteAllAndInsert(userId: user.id, evaluations: newEvaluations)
-                    } catch let error {
-                        logger?.error(error)
-                        completion?(.failure(error: .init(error: error), featureTag: featureTag))
-                        return
+            timeoutMillis: timeoutMillis,
+            condition: UserEvaluationCondition(
+                evaluatedAt: evaluatedAt,
+                userAttributesUpdated: userAttributesUpdated)) { [weak self] result in
+            switch result {
+            case .success(let response):
+                let newEvaluationsId = response.userEvaluationsId
+                if currentEvaluationsId == newEvaluationsId {
+                    logger?.debug(message: "Nothing to sync")
+                    completion?(result)
+                    return
+                }
+                let newEvaluations = response.evaluations.evaluations
+                self?.evaluations[user.id] = newEvaluations
+                self?.currentEvaluationsId = newEvaluationsId
+                // https://github.com/bucketeer-io/android-client-sdk/issues/69
+                // evaluatedAt: the last time the user was evaluated. The server will return in the get_evaluations response (UserEvaluations.CreatedAt), and it must be saved in the client
+                // userAttributesUpdated: when the user attributes change via the customAttributes interface, the userAttributesUpdated field must be set to true in the next request.
+                self?.stateStorage.setEvaluatedAt(value: response.evaluations.createdAt)
+                self?.stateStorage.setUserAttributesUpdated(value: false)
+
+                var shouldNotify = true
+                do {
+                    // forceUpdate: a boolean that tells the SDK to delete all the current data and save the latest evaluations from the response
+                    if response.evaluations.forceUpdate {
+                        try self?.forceUpdate(userId: user.id, evaluations: response.evaluations.evaluations)
```

Comment:
No need to create another function for this. Let's call it directly.

```suggestion
                        try evaluationDao.deleteAllAndInsert(userId: user.id, evaluations: response.evaluations.evaluations)
```

Also, add the comments and the `createdAt` to the update function so you can save it with the others after updating the evaluations in the DB

https://github.com/bucketeer-io/javascript-client-sdk/blob/main/src/internal/evaluation/EvaluationInteractor.ts#L47-L54

Reply:
I would like to keep it, because `forceUpdate` is more easily to understand. 
It also helps to quickly look up the place where the `forceUpdate` logic is when we view the method list of the class from the IDE window dropdown.